### ISSUE TITLE
RUF059 does not recognize variable string references like in pandas Dataframe.query

### ISSUE THREAD

**Issue by JanSellner** [astral-sh/ruff#20339]
### Summary

Unfortunately, RUF059 inherits the [same problem as F841](https://github.com/astral-sh/ruff/issues/12565) by not detecting string usages, especially in `df.query()`.

```python
import pandas as pd

def get_name():
    return "Alice", "Bob"

def main():
    df = pd.DataFrame({
        "name": ["Alice", "Bob", "Charlie"],
        "age": [25, 30, 35],
        "city": ["New York", "Los Angeles", "Chicago"]
    })

    name1, name2 = get_name()
    print(df.query("name == @name1"))
    print(df.query("name == @name2"))

main()
```

Applying the unsafe fix breaks the code obviously.

A full solution to this problem is probably hard to impossible because variables could be used in strings in every conceivable way. However, I still think that `df.query()` is a very common use case and catching that would already help a lot of users (and would also fix F841). Maybe this could even be generalized so that the user specifies function names (like `query` or `execute`) and then those strings will be searched for the variable name as well (of course, not a perfect solution but maybe good enough for most use cases?).

If you don't want to open this potential box of Pandora (understandable), then maybe at least a warning to the documentation on why this rule's fix is and will be unsafe?

**Comment by ntBre** [comment]
Thanks for the report! I think it could definitely be worth mentioning this in the docs for both rules, even if we don't end up changing the rules themselves.
